dom/devicestorage/DeviceStorageStatics.cpp

```diff
@@ -309,18 +309,18 @@ DeviceStorageStatics::GetDir(DeviceStorageType aType)
 
   nsCOMPtr<nsIFile> file;
   switch (aType) {
-    case TYPE_APPS:
-    case TYPE_CRASHES:
-    case TYPE_OVERRIDE:
-      file = sInstance->mDirs[aType];
-      return file.forget();
+    case TYPE_SDCARD:
+      // use overrideRootDir only for 'sdcard'
+      // see desktop.js
+      break;
     default:
+      file = sInstance->mDirs[aType];
+      if (file) {
+        return file.forget();
+      }
```

Comment:
The previous code was returning here. Here you only `break`. Is it what you wanted?